spec/index.bs

```diff
@@ -11283,6 +11353,72 @@ It must only be included by interfaces which also include those mixins.
     Implementations are strongly encouraged to optimize this algorithm.
 </div>
 
+## Immediate Data ## {#programmable-passes-immediate-data}
+
+<dl dfn-type=method dfn-for=GPUBindingCommandsMixin>
+    : <dfn>setImmediateData(rangeOffset, data, dataOffset, size)</dfn>
+    ::
+        Sets immediate data for subsequent render or compute commands.
+
+        <div algorithm=GPUBindingCommandsMixin.setImmediateData>
+            <div data-timeline=content>
+                **Called on:** {{GPUBindingCommandsMixin}} |this|.
+
+                **Arguments:**
+
+                <pre class=argumentdef for="GPUBindingCommandsMixin/setImmediateData(rangeOffset, data, dataOffset, size)">
+                    |rangeOffset|: Offset in bytes into the immediate data range to begin writing at.
+                        Must be a multiple of 4.
+                    |data|: Data to write into the immediate data range.
+                    |dataOffset|: Offset into |data| to begin writing from. Given in elements if
+                        |data| is a {{TypedArray}} and bytes otherwise. Defaults to 0.
+                    |size|: Size of content to write from |data|. Given in elements if
+                        |data| is a {{TypedArray}} and bytes otherwise.
+                </pre>
+
+                **Returns:** {{undefined}}
+
+                [=Content timeline=] steps:
+
+                1. If |data| is an {{ArrayBuffer}} or {{DataView}}, let the element type be "byte".
+                    Otherwise, |data| is a TypedArray; let the element type be the type of the TypedArray.
+                1. Let |dataSize| be the size of |data|, in elements.
+                1. If |size| is missing,
+                    let |contentsSize| be |dataSize| &minus; |dataOffset|.
+                    Otherwise, let |contentsSize| be |size|.
+                1. If any of the following conditions are unsatisfied,
+                    throw an {{OperationError}} and return.
+
+                    <div class=validusage>
+                        - |contentsSize| &ge; 0.
+                        - |dataOffset| + |contentsSize| &le; |dataSize|.
+                        - |rangeOffset| is a multiple of 4.
```

Comment:
I believe that rangeOffset being a multiple of 4 should be a device timeline check since it is not required to implement the content side validation of the ArrayBuffer.